src/dmd/mars.d

```diff
@@ -124,6 +124,23 @@ Where:
 %.*s", cast(int)inifileCanon.length, inifileCanon.ptr, cast(int)help.length, &help[0]);
 }
 
+/// DMD specific options that are separate from the frontend.
+struct DMDparams
+{
+    bool alwaysframe;       // always emit standard stack frame
+    bool map;               // generate linker .map file
+
+    // Hidden debug switches
+    bool debugb;
+    bool debugc;
+    bool debugf;
+    bool debugr;
+    bool debugx;
+    bool debugy;
+}
+
+shared DMDparams dmdParams;
```

Comment:
This won't work with DMD as library and will create yet another global to be careful about. Why not leave it in globals and version it for digitalmars?

Reply:
No it doesn't? You wouldn't be using the mars module in a library. And the places that depend on it certainly wouldn't be part of said library as well.